Gems/Atom/Feature/Common/Assets/Shaders/ScreenSpace/DeferredFog.azsl

```diff
@@ -78,33 +102,43 @@ float CalculateFogDensity(float3 surfacePosWS, float depthInLayer)
 
     return fogDensity * fogRelativeHeight;
 }
+#endif //ENABLE_FOG_LAYER
 
 // An optimized version of deferred fog with ability to have height layer.
 // It is optimized to reduce almost all edge cases by using ray intersection logic 
 PSOutput MainPS(VSOutput IN)
 {
     PSOutput OUT;
     
-    float linearDepth = PassSrg::m_linearDepthTexture.Sample(PassSrg::DepthSampler, IN.m_texCoord).r;
-    float zDepth = PassSrg::m_depthStencilTexture.Sample(PassSrg::DepthSampler, IN.m_texCoord).r;
-    float3 surfacePosWS = WorldPositionFromDepthBuffer(PassSrg::m_depthStencilTexture, zDepth, IN.m_position.xy).xyz;
-
-    // Fog layer deltas
-    const float layerThickness = PassSrg::m_fogMaxHeight - PassSrg::m_fogMinHeight;
-    const float dViewMax = ViewSrg::m_worldPosition.z - PassSrg::m_fogMaxHeight;
-    const float dViewMin = ViewSrg::m_worldPosition.z - PassSrg::m_fogMinHeight;
-    const float3 surfaceViewVec = surfacePosWS - ViewSrg::m_worldPosition;
-    const float3 surfaceViewDir = normalize(surfaceViewVec);
-    float dSurfaceView = surfaceViewVec.z;
-
-    // When set to 0, indicate that View and surface hit point are outside the 
-    // fog layer / ray start-end point
-    float fogTraversalRelLength = 0;    // length of the traversal within the fog layer / ray length
-    float distToFogLayer = 0;
-    float fogDensity = 1.0;            // accumulated fog density along the ray traversal
+    float linearDepth;
+
+#if HAS_LINEAR_DEPTH
+    linearDepth = PassSrg::m_linearDepthTexture.Sample(PassSrg::DepthSampler, IN.m_texCoord).r;
+#else
+    linearDepth = PassSrg::m_depthStencilTexture.Sample(PassSrg::PointSampler, IN.m_texCoord).r;
+    linearDepth = abs(((ViewSrg::GetFarZTimesNearZ()) / (ViewSrg::GetFarZMinusNearZ() * linearDepth - ViewSrg::GetFarZ()))); 
+#endif
 
+#if ENABLE_FOG_LAYER
     if (o_enableFogLayer)
     {
+        float zDepth = PassSrg::m_depthStencilTexture.Sample(PassSrg::DepthSampler, IN.m_texCoord).r;
+        float3 surfacePosWS = WorldPositionFromDepthBuffer(PassSrg::m_depthStencilTexture, zDepth, IN.m_position.xy).xyz;
+
+        // Fog layer deltas
+        const float layerThickness = PassSrg::m_fogMaxHeight - PassSrg::m_fogMinHeight;
```

Comment:
does it matter that we're using `float` instead of `real` throughout this file?

Reply:
It doesn't matter that much since the layer and noise feature were disabled for mobile.